code/modules/urist/modules/wfc/modules/deepmaint/hooks.dm

```diff
@@ -0,0 +1,31 @@
+/hook/death/proc/deepmaint_death_handler(var/mob/living/carbon/human/H, var/gibbed = FALSE)
+	var/mob/living/carbon/human/deadman = H
```

Comment:
IIRC, DM doesn't _actually_ validate input params, IIRC, so you can wind up runtiming by calling procs on a type that does not have them. 

The cast in L2 here is doing the heavy lifting of nulling the input out if it's not the expected type so that we can do null checks on it to catch both mis-typed and explicitly null inputs.

Reply:
Yep, DM doesn't validate types passed to parameters. But iirc, assigning like this _doesn't_ null on incorrect types (might require validation, but I'm pretty sure that's how DM works). Pretty sure DM is kinda dumb like that. You'd have to do a `istype(H)` to catch both nulls and incorrect types